File: expand-widget/src/main/java/com/zxn/widget/ExpandScrollView.java
package com.zxn.widget;

import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.widget.ScrollView;

/**
 * 可以控制是否可以滚动的ScrollView,默认不可以滚动
 * Created by zxn on 2020/3/13.
 */
public class ExpandScrollView extends ScrollView {

    /**
     * 控制是否可以进行滑动
     */
    private boolean mScrollEnabled;

    public ExpandScrollView(Context context) {
        super(context);
    }

    public ExpandScrollView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public ExpandScrollView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public boolean onTouchEvent(MotionEvent ev) {
        return mScrollEnabled && super.onTouchEvent(ev);
    }


    @Override
    public boolean onInterceptTouchEvent(MotionEvent ev) {
        return mScrollEnabled && super.onInterceptTouchEvent(ev);
    }

    public boolean isScrollEnabled() {
        return mScrollEnabled;
    }

    public void setScrollEnabled(boolean mScrollEnabled) {
        this.mScrollEnabled = mScrollEnabled;
    }
}

File: expand-widget/src/main/java/com/zxn/widget/ExpandScrollView.kt
package com.zxn.widget

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import android.widget.ScrollView

/**可以控制是否可以滚动的ScrollView,默认不可以滚动
 *  Updated by zxn on 2020.10.17.
 */
class ExpandScrollView : ScrollView {
    /**
     * 控制是否可以进行滑动
     */
    var isScrollEnabled = false

    constructor(context: Context?) : super(context) {}
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs) {}
    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {}

    override fun onTouchEvent(ev: MotionEvent): Boolean {
        return isScrollEnabled && super.onTouchEvent(ev)
    }

    override fun onInterceptTouchEvent(ev: MotionEvent): Boolean {
        return isScrollEnabled && super.onInterceptTouchEvent(ev)
    }
}